### PR TITLE
yara-x: init at 0.3.0

### PR DESCRIPTION
**Pull request by LeSuisse**
## Description of changes

Announcement blogpost:
https://virustotal.github.io/yara-x/blog/yara-is-dead-long-live-yara-x/

Related to https://github.com/NixOS/nixpkgs/issues/81418

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by LeSuisse** [comment]
@fabaff Do you want to be added to the maintainers of this one since you are the current maintainer of `yara`?

File: pkgs/by-name/ya/yara-x/package.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "yara-x";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "VirusTotal";
+    repo = "yara-x";
+    rev = "v${version}";
```
Comment (fabaff) [inline pkgs/by-name/ya/yara-x/package.nix:17]: r-ryantm will update it with its first update.

```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/ya/yara-x/package.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      --fish <($out/bin/yr completion fish) \
+      --zsh <($out/bin/yr completion zsh)
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = yara-x;
+  };
+
+  meta = {
+    description = "The pattern matching swiss knife for malware researchers, and everyone else. A rewrite of YARA in Rust.";
```
Comment (fabaff) [inline pkgs/by-name/ya/yara-x/package.nix:37]:
```suggestion
    description = "Tool to do pattern matching for malware research";
```

see https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes about the period.

**Review by fabaff — COMMENTED** [review]
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/313215#issuecomment-2122112579)
- [ ] all depending packages build
- [x] tested on x86_64

```txt
rule eicar_string {
    strings:
        $eicar = "X5O!P%@AP[4\\PZX54(P^)7CC)7}$EICAR-STANDARD-ANTIVIRUS-TEST-FILE!$H+H*" fullword ascii

    condition:
        all of them
}
```

```bash
$ ./results/yara-x/bin/yr scan /home/test /home/test
eicar_string /home/test/eicar.txt
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 49645 file(s) scanned in 1.1s. 1 file(s) matched.

```

Looks good to me beside the nitpicking.

**Review by fabaff — APPROVED** [review]

**Pull request merged**